vis/plotly.js

```diff
@@ -0,0 +1,220 @@
+var x_data = [];
+var y_data = [];
+var y2_data = [81, 91, 88, 123, 43, 154, 171, 106, 142, 185, 144, 78, 124, 53];
+var y3_data = [];
+
+d3.json('projects-extended.json').then((res) => {
+    Object.keys(res).forEach(year => {
+        const quarters = Object.keys(res[year]).reverse();
+        const quartersinfo = Object.values(res[year]).reverse();
+        for(var i = 0; i < Object.values(quarters).length; i++){
+            var quarter = year + ' ' + Object.values(quarters)[i];
+            x_data.push(quarter);
+            y_data.push(Object.keys(Object.values(quartersinfo)[i]).length-3);
+
+            var project = Object.values(Object.values(quartersinfo)[i]);
+            var count = 0;
+            for (var k = 0; k< y_data[count]; k++)
+            {
+                y3_data.push(project[k]);
+            }
+            count++;
+        }
+    });
+
+    x_data.unshift('2016 Fall', '2017 Winter');
+    y_data.unshift(0, 0);
+
+    var trace1 =
+    {
+        x: x_data,
+        y: y_data,
+        yaxis: 'y1',
+        name: 'Number of Projects',
+        type: 'bar',
+        marker: {
+            color: 'rgb(158,202,225)',
+            opacity: 0.6,
+        },
+    };
+
+    var trace2 =
+    {
+          x: x_data,
+          y: y2_data,
+          yaxis: 'y2',
+          name: 'Number of Applicants',
+          type: 'scatter',
+          marker: {
+            color: 'rgb(255,0,0)',
+            opacity: 0.6,
+            line: {
+              color: 'rgb(255, 0, 0)',
+              width: 1.5,
+            },
+          },
+    };
+
+    var layout = {
+        title: 'Number of Applicants vs Number of Projects',
+        xaxis: {
+            title: 'Timeline',
+        },
+        yaxis: {
+            title: 'Number of Projects',
+        },
+        yaxis2: {
+            title: 'Number of Applicants',
+            overlaying: 'y',
+            side: 'right',
+        },
+    };
+
+    var alldata = [trace1, trace2];
+
+    Plotly.newPlot('apps_proj', alldata, layout);
+ });
+
+ var year_info = [];
+ var major_info = [];
+ var year_clean = [];
+ var major_clean = [];
+ var y_year = [0,0,0,0,0];
+ var y_major = [0,0,0,0,0,0,0,0,0,0,0];
+ d3.json('sorted-projectmember.json').then((res) => {
+    Object.keys(res).forEach(applicant => {
+        const applicantinfo = Object.values(res[applicant]);
+        year_info.push((applicantinfo)[1]);
+        year_info.push((applicantinfo)[0]);
+        major_info.push((applicantinfo)[0]);
+    });
+    for (index in year_info) {
+        var info = year_info[index];
+        if(info=='1' || info=='1st' || info=='2' || info=='2nd' ||
+        info=='3' || info=='3rd' || info=='4' || info=='4+' || info=='4th' || info=='4th+') {
```

Comment:
u can simpliy this if statement by making an array of valid elements and using the `Array.includes()`
```
const VALID_YEARS = ['1', '1st', '2', '2nd', '3', '3rd', '4', '4+', ...];
if (VALID_YEARS.includes(info)) {
  ...
}
```

Reply:
Also, i would consider just removing the letters from the words and that way you don't have to do a switch statement later!

i.e. find a way to remove the `st` in `1st` and `nd` in `2nd` (hint use the `string.val().replace()` function with regex expresssion